src/Features/CSharp/Portable/ReplaceDiscardDeclarationsWithAssignments/CSharpReplaceDiscardDeclarationsWithAssignmentsService.cs

```diff
@@ -21,21 +22,72 @@ namespace Microsoft.CodeAnalysis.CSharp.ReplaceDiscardDeclarationsWithAssignment
     [ExportLanguageService(typeof(IReplaceDiscardDeclarationsWithAssignmentsService), LanguageNames.CSharp), Shared]
     internal sealed class CSharpReplaceDiscardDeclarationsWithAssignmentsService : IReplaceDiscardDeclarationsWithAssignmentsService
     {
-        public Task<SyntaxNode> ReplaceAsync(SyntaxNode memberDeclaration, CancellationToken cancellationToken)
+        public Task<SyntaxNode> ReplaceAsync(SyntaxNode memberDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
             var editor = new SyntaxEditor(memberDeclaration, CSharpSyntaxGenerator.Instance);
             foreach (var child in memberDeclaration.DescendantNodes())
             {
-                if (child is LocalDeclarationStatementSyntax localDeclarationStatement &&
-                    localDeclarationStatement.Declaration.Variables.Any(IsDiscardDeclaration))
+                switch (child)
                 {
-                    RemoveDiscardHelper.ProcessDeclarationStatement(localDeclarationStatement, editor);
-                }
-                else if (child is CatchDeclarationSyntax catchDeclaration &&
-                    IsDiscardDeclaration(catchDeclaration))
-                {
-                    // "catch (Exception _)" => "catch (Exception)"
-                    editor.ReplaceNode(catchDeclaration, catchDeclaration.WithIdentifier(default));
+                    case LocalDeclarationStatementSyntax localDeclarationStatement:
+                        if (localDeclarationStatement.Declaration.Variables.Any(IsDiscardDeclaration))
+                        {
+                            RemoveDiscardHelper.ProcessDeclarationStatement(localDeclarationStatement, editor);
+                        }
+
+                        break;
+
+                    case CatchDeclarationSyntax catchDeclaration:
+                        if (IsDiscardDeclaration(catchDeclaration))
+                        {
+                            // "catch (Exception _)" => "catch (Exception)"
+                            editor.ReplaceNode(catchDeclaration, catchDeclaration.WithIdentifier(default));
```

Comment:
is the final result formatted?  Otherwise, does this end up leaving the space after the `Exception`?

Reply:
Yes, test over here: https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/CSharpTest/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs#L3797